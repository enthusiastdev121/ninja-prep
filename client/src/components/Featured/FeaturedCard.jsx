import React from 'react'
import './featured.css'

const FeaturedCard = (props) => {
    return (
        <div className="card ">
            <img className="card-img-top" src={props.discoverCardImgPath} alt="Card cap" />
            <div className="card-body">
                <h2 className="card-title">{props.discoverCardTitle}</h2>
                <p className="card-text subtitle">{props.discoverCardSubtitle}</p>
            </div>
        </div>
    )
}

export default FeaturedCard
export { default as FeaturedCard } from './FeaturedCard'
