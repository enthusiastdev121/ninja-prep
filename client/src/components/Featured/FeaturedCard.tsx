import React from 'react'
import './featured.css'

interface Props {
    discoverCardImgPath: string
    discoverCardTitle: string
}

const FeaturedCard = (props: Props) => {
    return (
        <div className="card mr-5">
            <img className="card-img-top img-fluid" src={props.discoverCardImgPath} alt="Card cap" />
            <div className="card-body">
                <p className="card-title lead text-dark">{props.discoverCardTitle}</p>
            </div>
        </div>
    )
}

export default FeaturedCard
