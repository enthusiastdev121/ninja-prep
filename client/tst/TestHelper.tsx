import React from 'react'
import { Router } from 'react-router-dom'
import { createBrowserHistory } from 'history'
const history = createBrowserHistory()

interface DEVICE_WIDTHS {
    readonly DESKTOP: number
    readonly MOBILE: number
    readonly TABLET: number
}

//Widths in Pixels. Aligns with Media Query widths
export const DEVICE_TYPES: DEVICE_WIDTHS = {
    DESKTOP: 1000,
    MOBILE: 750,
    TABLET: 800
}

export const routeWrapper = (children: React.ReactChild) => {
    return <Router history={history}>{children}</Router>
}
