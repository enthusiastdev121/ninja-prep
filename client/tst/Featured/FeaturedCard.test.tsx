import React from 'react'
import FeaturedCard from '../../src/components/Featured/FeaturedCard'
import { shallow, mount } from 'enzyme'
import { Context as ResponsiveContext } from 'react-responsive'
import { DEVICE_TYPES, routeWrapper } from '../TestHelper'
import toJson from 'enzyme-to-json'

test.each(Object.entries(DEVICE_TYPES))('renders successfully on %p', (deviceType: string, deviceWidth: number) => {
    const wrappingComponent = ResponsiveContext.Provider
    const wrappingComponentProps = { value: { width: deviceWidth } }
    const mountProps = { wrappingComponent, wrappingComponentProps }

    const wrapper = shallow(routeWrapper(<FeaturedCard discoverCardImgPath="" discoverCardTitle="" />), mountProps)
    expect(wrapper).toBeDefined
})

test.each(Object.entries(DEVICE_TYPES))('matches snapshot on %p', (deviceType: string, deviceWidth: number) => {
    const wrappingComponent = ResponsiveContext.Provider
    const wrappingComponentProps = { value: { width: deviceWidth } }
    const mountProps = { wrappingComponent, wrappingComponentProps }

    const wrapper = mount(routeWrapper(<FeaturedCard discoverCardImgPath="" discoverCardTitle="" />), mountProps)
    expect(toJson(wrapper)).toMatchSnapshot()
})
