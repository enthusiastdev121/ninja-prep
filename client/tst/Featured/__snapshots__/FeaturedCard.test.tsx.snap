// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot on "DESKTOP" 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <FeaturedCard
    discoverCardImgPath=""
    discoverCardTitle=""
  >
    <div
      className="card mr-5"
    >
      <img
        alt="Card cap"
        className="card-img-top img-fluid"
        src=""
      />
      <div
        className="card-body"
      >
        <p
          className="card-title lead text-dark"
        />
      </div>
    </div>
  </FeaturedCard>
</Router>
`;

exports[`matches snapshot on "MOBILE" 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <FeaturedCard
    discoverCardImgPath=""
    discoverCardTitle=""
  >
    <div
      className="card mr-5"
    >
      <img
        alt="Card cap"
        className="card-img-top img-fluid"
        src=""
      />
      <div
        className="card-body"
      >
        <p
          className="card-title lead text-dark"
        />
      </div>
    </div>
  </FeaturedCard>
</Router>
`;

exports[`matches snapshot on "TABLET" 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <FeaturedCard
    discoverCardImgPath=""
    discoverCardTitle=""
  >
    <div
      className="card mr-5"
    >
      <img
        alt="Card cap"
        className="card-img-top img-fluid"
        src=""
      />
      <div
        className="card-body"
      >
        <p
          className="card-title lead text-dark"
        />
      </div>
    </div>
  </FeaturedCard>
</Router>
`;
